target/riscv/monitor.c

```diff
@@ -64,6 +64,42 @@ static void print_pte(Monitor *mon, int va_bits, target_ulong vaddr,
         return;
     }
 
+#if defined(TARGET_CHERI) && !defined(TARGET_RISCV32)
+    char cr, cw;
+
+    switch (attr & (PTE_CW | PTE_CD)) {
+    case 0: // trap
+        cw = '-';
+        break;
+    case PTE_CD:
+    case PTE_CW | PTE_CD: // store OK
+        cw = 'S';
+        break;
+    case PTE_CW: // store will dirty
+        cw = 's';
+        break;
+    }
```

Comment:
This should just be handled like W and D now with separate output for each.

Reply:
I did the same for the CR{,M,G} bits, too.